### ISSUE TITLE
Creating presigned request to S3 object lambda creates invalid signature

### ISSUE THREAD

**Issue by Petah** [aws/aws-sdk-php#2328]
Confirm by changing [ ] to [x] below to ensure that it's a bug:
- [x] I've gone though [Developer Guide](https://docs.aws.amazon.com/sdk-for-php/v3/developer-guide/welcome.html) and [API reference](https://docs.aws.amazon.com/aws-sdk-php/v3/api/index.html)
- [x] I've checked [AWS Forums](https://forums.aws.amazon.com) and [StackOverflow](https://stackoverflow.com/questions/tagged/aws-php-sdk) for answers
- [x] I've searched for [previous similar issues](https://github.com/aws/aws-sdk-php/issues) and didn't find any solution
  
**Describe the bug**
Creating a presigned URL to a object in a S3 Object Lambda ARN creates an invalid signature.

**To Reproduce (observed behavior)**
```php
$s3Client->createPresignedRequest($s3Client->getCommand('GetObject', [
    'Bucket' => 'arn:aws:s3-object-lambda:us-east-1:9999999999:accesspoint/foo-bar',
    'Key' => 'path/to/object.txt',
]), 86400)
```
It creates a URL like this:

https://foo-bar-9999999999.s3-object-lambda.us-east-1.amazonaws.com/path/to/object.txt?X-Amz-Content-Sha256=UNSIGNED-PAYLOAD&X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKXXXXXXXXXXXXXXXXXX%2F20211005%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20211005T192250Z&X-Amz-SignedHeaders=host&X-Amz-Expires=-1633375370&X-Amz-Signature=68bcb40fb6d00f851d697940cb9130792d4104d7bebe739dcb614585c00b9a3d

**Expected behavior**
But should be this (note the service in the X-Amz-Credential param is s3 above, but should be s3-object-lambda)

https://foo-bar-9999999999.s3-object-lambda.us-east-1.amazonaws.com/path/to/object.txt?X-Amz-Content-Sha256=UNSIGNED-PAYLOAD&X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKXXXXXXXXXXXXXXXXXX%2F20211005%2Fus-east-1%2Fs3-object-lambda%2Faws4_request&X-Amz-Date=20211005T192250Z&X-Amz-SignedHeaders=host&X-Amz-Expires=-1633375370&X-Amz-Signature=68bcb40fb6d00f851d697940cb9130792d4104d7bebe739dcb614585c00b9a3d

**Version of AWS SDK for PHP?**
3.197.0

**Version of PHP (`php -v`)?**
8.0.9

**Comment by dencorg** [comment]
I can confirm I encounter the same InvalidSignature error when trying to get an object from an s3 object lambda access point.

The same call works using the node sdk.

**Comment by Petah** [comment]
I did manage to get it working doing this:

```
$options = [
    'Bucket' => $objectLambdaArn,
    'Key' => $key,
];
[$expiry, $startTime] = $this->getRoundedExpiryTime();
$command = $this->getS3Client()->getCommand('GetObject', $options);
$command->getHandlerList()->appendBuild(Middleware::mapRequest(function (RequestInterface $request) use ($customQueryParams) {
    return $request->withUri($request->getUri()->withQuery(trim($request->getUri()->getQuery() . '&' . http_build_query([
        'foo-payload' => json_encode($customQueryParams, JSON_UNESCAPED_SLASHES)
    ]), '&')));
}), 'add-custom-param');
$signer = new S3SignatureV4('s3-object-lambda', 'us-east-1', ['unsigned-body' => 'true']);
$request = \Aws\serialize($command);
return (string) $signer->presign($request, $this->getS3Client()->getCredentials()->wait(), $expiry, [
    'start_time' => $startTime,
])->getUri();
```

Main part being the `new S3SignatureV4('s3-object-lambda',...`, but this also adds some extra custom params into the URL etc.

**Comment by yenfryherrerafeliz** [comment]
Hi @Petah, thanks for bringing this to our attention. I was able to reproduce the reported behavior and I was also able to confirm that by adding s3-object-lambda instead of s3 in the X-Amz-Credential param fixes the problem. I will mark this issue to review so it can be further addressed.

Thanks.

**Comment by stobrien89** [comment]
This should be resolved by #2459, which will be available in tomorrow's release.  Let us know you have any questions!
